sql/querycache.go

```diff
@@ -0,0 +1,143 @@
+package sql
+
+import (
+	"slices"
+	"sync"
+)
+
+type QueryCacheKey struct {
+	Kind string
+	Key  string
+}
+
+func MkQueryCacheKey(kind, key string) QueryCacheKey {
```

Comment:
NIT: In my opinion the function could be called `QueryCacheKey` and the type something different (the type doesn't even need to be exported).

Reply:
Fixed